### PR TITLE
drainer/translater: fix the bug that drainer writes incorrect time data

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb-binlog/issues/1159

### What is changed and how it works?
Adjust drainer's timestamp data to downstream's timezone for mysql/tidb.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

- None

Related changes

 - Need to cherry-pick to the release branch

### Release note

- fix the bug that drainer writes incorrect time data when drainer and downstream database are not in a seem time zone

### CODE REVIEW

**Comment by lance6716** [comment]
test failed

**Comment by lichunzhu** [comment]
> test failed

Integration test failure is caused by `daylight time` problem...

File: drainer/sync/kafka.go [hunk 1]
```diff
@@ -22,10 +22,11 @@ import (
 	"github.com/Shopify/sarama"
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
-	"github.com/pingcap/tidb-binlog/drainer/translator"
-	"github.com/pingcap/tidb-binlog/pkg/util"
 	obinlog "github.com/pingcap/tidb-tools/tidb-binlog/proto/go-binlog"
 	"go.uber.org/zap"
+
+	"github.com/pingcap/tidb-binlog/drainer/translator"
```
Comment (hi-rustin) [inline drainer/sync/kafka.go:28]: Why we need this order change? Could you please revert it?

Comment (lichunzhu) [reply drainer/sync/kafka.go:28]: It's auto-adjusted by fmt. Most directory import order in tidb-binlog is like this.

**Comment by lichunzhu** [comment]
@lance6716 PTAL again

File: drainer/pump.go [hunk 2]
```diff
@@ -176,6 +177,7 @@ func (p *Pump) PullBinlog(pctx context.Context, last int64) chan MergeItem {
 				p.reportErr(pctx, err)
 				return
 			}
+			resp.Entity.Payload = resp.Entity.Payload[:0]
```
Comment (lance6716) [inline drainer/pump.go:180]: what's the purpose of it

File: tests/binlog/run.sh [hunk 3]
```diff
@@ -4,6 +4,16 @@ set -e
...
 cd "$(dirname "$0")"
 
+function revert_timezone() {
+    run_sql "SET @@global.time_zone = 'SYSTEM';"
+    down_run_sql "SET @@global.time_zone = 'SYSTEM';"
+}
+
+# set timezone to others before drainer starts
+trap revert_timezone EXIT
+run_sql "SET @@global.time_zone = 'Asia/Shanghai';"
```
Comment (lance6716) [inline tests/binlog/run.sh:14]: maybe set another value? system timezone is likely to be +8

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: drainer/pump.go (hunk 2 above)
Comment (lichunzhu) [reply drainer/pump.go:180]: to make sure this array can be GC asap

Comment (lance6716) [reply drainer/pump.go:180]: maybe resp.Entity.Payload = nil . I'm afraid a zero length slice will still hold the reference to underlying array

**Review by lance6716 — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by gozssky — APPROVED** [review]

**Pull request merged**